pkg/database/mobile_app.go

```diff
@@ -0,0 +1,93 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package database
+
+import (
+	"github.com/jinzhu/gorm"
+)
+
+type OSType int
+
+const (
+	OSTypeIOS OSType = iota
+	OSTypeAndroid
+)
+
+type MobileApp struct {
+	gorm.Model
+	Errorable
+
+	// Name is the name of the app.
+	Name string `gorm:"name,varchar(512);unique_index:realm_app_name"`
+
+	// RealmID is the id of the mobile app.
+	RealmID uint `gorm:"unique_index:realm_app_name"`
+
+	// OS is the type of the application we're using (eg, iOS, Android).
+	OS OSType `gorm:"os,type:int"`
+
+	// IOSAppID is a unique string representing the app.
+	AppID string `gorm:"app_id,type:varchar(512)"`
+
+	// SHA is a unique hash of the app.
+	// It is only present for Android devices, and should be of the form:
+	//   AA:BB:CC:DD...
+	SHA string `gorm:"sha,type:text"`
+}
+
+// FindActiveAppsByOS finds all authorized by their OS.
+func (db *Database) FindActiveAppsByOS(os OSType) ([]*MobileApp, error) {
+	// Find the API key that matches the constraints.
+	var apps []*MobileApp
+	if err := db.db.
+		Model(&MobileApp{}).
+		Where("os = ?", os).
+		Where("deleted_at IS NULL").
+		Find(&apps).
+		Error; err != nil {
+		return nil, err
+	}
+	return apps, nil
+}
+
+// CreateMobileApp adds a mobile app to the DB.
+func (r *Realm) CreateMobileApp(db *Database, app *MobileApp) error {
```

Comment:
I'm not sure we want mutating functions to live off other database models. This should probably live off of `Database`.